### PR TITLE
PWGHF: Implement tree creator for B0

### PR DESCRIPTION
**Pull request by AlexBigO**
Hi @fgrosa, here is the tree creator for B0. Basically a copy of the D+ one, without `prong2` and with only `NSigma` for `prong1`.

### CODE REVIEW

**Review by fgrosa — COMMENTED** [review]
Ciao @AlexBigO thanks a lot!
I would only add the product of impact parameters, i.e. https://github.com/AliceO2Group/O2Physics/blob/b0806953b839b003b7cde26ff67f5befef72b69b/PWGHF/DataModel/CandidateReconstructionTables.h#L1852
which should be rather powerful for all B mesons.

File: PWGHF/TableProducer/treeCreatorB0ToDPi.cxx [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+  Produces<o2::aod::HfCandB0FullParticles> rowCandidateFullParticles;
+  Produces<o2::aod::HfCandB0Lite> rowCandidateLite;
+
+  Configurable<int> selectionFlagB0{"selectionB0", 1, "Selection Flag for B0"};
+  Configurable<bool> fillCandidateLiteTable{"fillCandidateLiteTable", false, "Switch to fill lite table with candidate properties"};
+
+  // parameters for production of training samples
+  Configurable<bool> fillOnlySignal{"fillOnlySignal", false, "Flag to fill derived tables with signal for ML trainings"};
+  Configurable<bool> fillOnlyBackground{"fillOnlyBackground", false, "Flag to fill derived tables with background for ML trainings"};
+  Configurable<float> donwSampleBkgFactor{"donwSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorB0ToDPi.cxx:170]:
```suggestion
  Configurable<float> downSampleBkgFactor{"downSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Comment by vkucera** [comment]
Hi @AlexBigO , what is the status of this PR?

**Comment by AlexBigO** [comment]
Hi @fgrosa , @vkucera . Thanks for the comments.

File: PWGHF/TableProducer/treeCreatorB0ToDPi.cxx [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        candidate.chi2PCA(),
+        candidate.decayLength(),
+        candidate.decayLengthXY(),
+        candidate.decayLengthNormalised(),
+        candidate.decayLengthXYNormalised(),
+        candidate.ptProng0(),
+        candidate.ptProng1(),
+        candidate.impactParameter0(),
+        candidate.impactParameter1(),
+        candidate.impactParameter0() * candidate.impactParameter1(),
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorB0ToDPi.cxx:219]: @AlexBigO @fgrosa , how is it useful to save both impact parameters and also their product?

Comment (fgrosa) [reply PWGHF/TableProducer/treeCreatorB0ToDPi.cxx:219]: Indeed, we can spare disk space and compute it offline if needed. Sorry for the previous comment!

File: PWGHF/TableProducer/treeCreatorB0ToDPi.cxx [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+/// \brief Writer of B0 → D- π+ candidates in the form of flat tables to be stored in TTrees.
+///        Intended for debug, local optimization of analysis on small samples or ML training.
+///        In this file are defined and filled the output tables
+///
+/// \author Alexandre Bigot <alexandre.bigot@cern.ch>, IPHC Strasbourg
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "PWGHF/DataModel/CandidateReconstructionTables.h"
+#include "PWGHF/DataModel/CandidateSelectionTables.h"
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorB0ToDPi.cxx:22]:
```suggestion
#include "Framework/AnalysisTask.h"
#include "Framework/runDataProcessing.h"

#include "PWGHF/DataModel/CandidateReconstructionTables.h"
#include "PWGHF/DataModel/CandidateSelectionTables.h"
```

File: PWGHF/TableProducer/treeCreatorB0ToDPi.cxx [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+  Configurable<bool> fillCandidateLiteTable{"fillCandidateLiteTable", false, "Switch to fill lite table with candidate properties"};
+
+  // parameters for production of training samples
+  Configurable<bool> fillOnlySignal{"fillOnlySignal", false, "Flag to fill derived tables with signal for ML trainings"};
+  Configurable<bool> fillOnlyBackground{"fillOnlyBackground", false, "Flag to fill derived tables with background for ML trainings"};
+  Configurable<float> downSampleBkgFactor{"downSampleBkgFactor", 1., "Fraction of background candidates to keep for ML trainings"};
+  Configurable<float> ptMaxForDownSample{"ptMaxForDownSample", 10., "Maximum pt for the application of the downsampling factor"};
+
+  Filter filterSelectCandidates = aod::hf_sel_candidate_b0::isSelB0ToDPi >= selectionFlagB0;
+  using SelectedCandidatesMc = soa::Filtered<soa::Join<aod::HfCandB0, aod::HfCandB0McRec, aod::HfSelB0ToDPi>>;
```
Comment (vkucera) [inline PWGHF/TableProducer/treeCreatorB0ToDPi.cxx:174]:
```suggestion
  using SelectedCandidatesMc = soa::Filtered<soa::Join<aod::HfCandB0, aod::HfCandB0McRec, aod::HfSelB0ToDPi>>;

  Filter filterSelectCandidates = aod::hf_sel_candidate_b0::isSelB0ToDPi >= selectionFlagB0;
```

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by vkucera — APPROVED** [review]

**Review by vkucera — CHANGES_REQUESTED** [review]

**Review by vkucera — APPROVED** [review]

**Pull request merged**